WordPress/WordPressTest/PostRepositorySaveTests.swift

```diff
@@ -0,0 +1,1059 @@
+            }()])
+            post.isStickyPost = true
+
+            return revision
+        }()
+
+        XCTAssertNotNil(post.revision, "Revision is missing")
+
+        // GIVEN a server where the post
+        stub(condition: isPath("/rest/v1.2/sites/80511/posts/974")) { request in
+            // THEN the app sends a partial update
+            try assertRequestBody(request, expected: """
+            {
+              "author" : 948,
+              "categories_by_id" : [
+                53
+              ],
+              "content" : "content-b",
+              "date" : "2024-03-07T23:00:50+0000",
+              "excerpt" : "excerpt-b",
+              "featured_image" : 92,
+              "if_not_modified_since" : "2024-03-07T23:00:40+0000",
+              "password" : "123",
+              "slug" : "slug-b",
+              "status" : "publish",
+              "sticky" : true,
+              "title" : "title-b"
+            }
+            """)
+
+            var post = WordPressComPost.mock
+            post.title = "title-b"
+            return try HTTPStubsResponse(value: post, statusCode: 202)
+        }
+
+        // WHEN saving the post
+        try await repository._save(post)
+
+        // THEN the title got updated
+        XCTAssertEqual(post.postTitle, "title-b")
+        // THEN the local revision got deleted
+        XCTAssertNil(post.revision)
+        XCTAssertNil(revision.managedObjectContext)
+    }
+
+    /// Scenario: user edits a post that got trashed on the backend.
+    func testSaveExistingPostTrashedOnServer() async throws {
+        // GIVEN a draft post (synced)
+        let post = makePost {
+            $0.status = .draft
+            $0.postID = 974
+            $0.authorID = 29043
+            $0.dateCreated = Date(timeIntervalSince1970: 1709852440)
+            $0.dateModified = Date(timeIntervalSince1970: 1709852440)
+            $0.postTitle = "Hello"
+            $0.content = "content-a"
+        }
+
+        // GIVEN a local revision with an updated title
+        let revision = post.createRevision()
+        revision.postTitle = "title-b"
+
+        XCTAssertNotNil(post.revision, "Revision is missing")
+
+        // GIVEN a server where the post
+        stub(condition: isPath("/rest/v1.2/sites/80511/posts/974")) { request in
+            // THEN the app sends a partial update
+            try assertRequestBody(request, expected: """
+            {
+              "title" : "title-b"
+            }
+            """)
+
+            var post = WordPressComPost.mock
+            post.status = Post.Status.trash.rawValue
+            post.title = "title-b"
+            return try HTTPStubsResponse(value: post, statusCode: 202)
+        }
+
+        // WHEN saving the post
+        try await repository._save(post)
+
+        // THEN post got updated and the status now reflects the status on backend
+        XCTAssertEqual(post.status, .trash)
+        XCTAssertEqual(post.postTitle, "title-b")
```

Comment:
I'm slightly wondering if it makes sense to accept changes for a post that was trashed on remote.